src/libisyntax.h

```diff
@@ -1,3 +1,68 @@
 #pragma once
 
-void libisyntax_init();
+#include <stdint.h>
+
+// API conventions:
+// - return type is one of:
+// -- appropriate data type for simple getters. For complex getters doing work, isyntax_error_t + out variable.
+// -- void if the function doesn't fail at runtime (e.g. destructors).
+// -- isyntax_error_t otherwise.
+// - Output arguments are last, name prefixed with 'out_'.
+// - 'Object operated on' argument is first.
+// - Function names are prefixed with 'libisyntax_', as C doesn't have namespaces. Naming convention
+//   is: libisyntax_<object>_<action>(..), where <object> is omitted for the object representing the isyntax file.
+// - Functions don't check for null/out of bounds, for efficiency. (they may assert, but that is implementation detail).
+// - Booleans are represented as int32_t and prefxied with 'is_' or 'has_'.
+// - Const applied to pointers is used as a signal that the object will not be modified.
+// - Prefer int even for unsigned types, see java rationale.
+
+
+typedef int32_t isyntax_error_t;
+
+#define LIBISYNTAX_OK 0
+// Generic error that the user should not expect to recover from.
+#define LIBISYNTAX_FATAL 1
+// One of the arguments passed to a function is invalid.
+#define LIBISYNTAX_INVALID_ARGUMENT 2
```

Comment:
Could possibly be an enum for better signaling through the type system.

Reply:
In C, enum size is not defined afaik. In ARCore API, I had to do the following tricks to make it work:
https://github.com/google-ar/arcore-android-sdk/blob/master/libraries/include/arcore_c_api.h#L665

We can do it here, I added a TODO in my code.

Reply:
Eh, I think it's not worth carrying a workaround.  An `int32_t` is okay.